src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeAtClauseBlockCheck.java

```diff
@@ -0,0 +1,197 @@
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">
+ * Tight-HTML Rules</a>.
+ * Type is {@code boolean}. Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;RequireEmptyLineBeforeAtClauseBlock&quot;/&gt;
+ * </pre>
+ * <p>
+ * By default, the check will report a violation if there is no blank line before the at-clause,
+ * like in the example below.
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * testMethod's javadoc.
+ *  * &#64;return something (violation)
+ *  *&#47;
+ * public boolean testMethod() {
+ *     return false;
+ * }
+ * </pre>
+ * <p>
+ *  Valid javadoc should have a blank line separating the parameters, return, throw, or
+ *  other at-clauses like in the example below.
+ *  </p>
+ *  <pre>
+ *  &#47;**
+ *  * testMethod's javadoc.
+ *  *
+ *  * &#64;param firstParam
+ *  * &#64;return something
+ *  *&#47;
+ *  public boolean testMethod(int firstParam) {
+ *      return false;
+ *  }
+ *  </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.line.before}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.35
+ */
+@StatelessCheck
+public class RequireEmptyLineBeforeAtClauseBlockCheck extends AbstractJavadocCheck {
+
+    /**
+     * The key in "messages.properties" for the message that describes an at-clause in javadoc
+     * requiring an empty line before it.
+     */
+    public static final String MSG_JAVADOC_TAG_LINE_BEFORE = "javadoc.tag.line.before";
+
+    /**
+     * When javadoc has only a javadoc tag like {@literal @} in it, the JAVADOC_TAG in a JAVADOC
+     * detail node will always have 3 siblings before it. The parse tree looks like:
+     * <pre>
+     * JAVADOC
+     * \
+     *  -- NEWLINE
+     *  -- LEADING_ASTERISK
+     *  -- WS
+     *  -- JAVADOC_TAG
+     * </pre>
+     */
+    private static final int NUM_OF_SIBLINGS_BEFORE_ONLY_JAVADOC_TAG = 3;
+
+    @Override
+    public int[] getDefaultJavadocTokens() {
+        return new int[] {
+            JavadocTokenTypes.JAVADOC_TAG,
+        };
+    }
+
+    @Override
+    public int[] getRequiredJavadocTokens() {
+        return getAcceptableJavadocTokens();
+    }
+
+    /**
+     * Logs when there is no empty line before the at clause.
+     *
+     * @param atClause the at at-clause node to check for an empty space before it.
+     */
+    @Override
+    public void visitJavadocToken(DetailNode atClause) {
```

Comment:
please rename all `atClause` to `tag` in method names also.

Internally we call it tag https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#JAVADOC_TAG , so implementation should use same terminology.

I will review later if we need to change name Check  of Check, it is could be more user friendly, I need to refresh my mind on when `at-clause` is better to use.

Reply:
unfortunately name of Check need to changed, please read https://github.com/checkstyle/checkstyle/issues/6192#issuecomment-663861910 for more details.

new name `RequireEmptyLineBeforeBlockTagsGroupCheck.java `

Reply:
Ok. I renamed all the classes to follow `RequireEmptyLineBeforeBlockTagsGroup`. I also grepped for all occurrences of `atclause`, `at-clause`, and `at clause` and replaced it with tag to keep it consistent.

Reply:
Please move all that changes to separate PR, we already have issue on it https://github.com/checkstyle/checkstyle/issues/8552 .
Let's keep this PR simple, to speedup acceptance of it.